### ISSUE TITLE
False positive on rule `lines-between-class-memebers` if a comment occurs between class members

### ISSUE THREAD

**Issue by jrencz** [eslint/eslint#9665]
**Tell us about your environment**

* **ESLint Version:** 4.12
* **Node Version:** 8.9.1
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**
n/a

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "script",
        "ecmaFeatures": {}
    },
    "rules": {
        "lines-between-class-members": 2
    },
    "env": {}
}
```
(generated by [ESLint Demo](https://eslint.org/demo/#eyJ0ZXh0IjoiY2xhc3MgRm9vIHtcbiAgLyogZXNsaW50LWRpc2FibGUgY2xhc3MtbWV0aG9kcy11c2UtdGhpcyAqL1xuICAvKipcbiAgICogQHJldHVybnMge0FycmF5fVxuICAgKi9cbiAgZ2V0Rm9vKCkge1xuICAgIHRocm93IG5ldyBFcnJvcignQ2hpbGQgY2xhc3Mgc2hvdWxkIHJlaW1wbGVtZW50IHRoaXMgbWV0aG9kJyk7XG4gIH1cbiAgLyogZXNsaW50LWVuYWJsZSBjbGFzcy1tZXRob2RzLXVzZS10aGlzICovXG5cbiAgLyoqXG4gICAqIEByZXR1cm5zIHt1bmRlZmluZWR9XG4gICAqL1xuICBkb1NvbWV0aGluZygpIHt9XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImxpbmVzLWJldHdlZW4tY2xhc3MtbWVtYmVycyI6Mn0sImVudiI6e319fQ==))

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I have a class which purpose is to be extended by other classes in order to provide actual implementation of some methods.
The class has placeholder methods that only throw, but don't use context (the subclass implementation will)

Since this is an exception i wrapped the method with `/* eslint-disable class-methods-use-this */` just before the doc block and `/* eslint-enable class-methods-use-this */` just after the method body like that:

```js
class Foo {
  /* eslint-disable class-methods-use-this */
  /**
   * @returns {Array}
   */
  getFoo() {
    throw new Error('Child class should reimplement this method');
  }
  /* eslint-enable class-methods-use-this */

  /**
   * @returns {undefined}
   */
  doSomething() {}
}
```

[Demo showing false positive (demo1)](https://eslint.org/demo/#eyJ0ZXh0IjoiY2xhc3MgRm9vIHtcbiAgLyogZXNsaW50LWRpc2FibGUgY2xhc3MtbWV0aG9kcy11c2UtdGhpcyAqL1xuICAvKipcbiAgICogQHJldHVybnMge0FycmF5fVxuICAgKi9cbiAgZ2V0Rm9vKCkge1xuICAgIHRocm93IG5ldyBFcnJvcignQ2hpbGQgY2xhc3Mgc2hvdWxkIHJlaW1wbGVtZW50IHRoaXMgbWV0aG9kJyk7XG4gIH1cbiAgLyogZXNsaW50LWVuYWJsZSBjbGFzcy1tZXRob2RzLXVzZS10aGlzICovXG5cbiAgLyoqXG4gICAqIEByZXR1cm5zIHt1bmRlZmluZWR9XG4gICAqL1xuICBkb1NvbWV0aGluZygpIHt9XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImxpbmVzLWJldHdlZW4tY2xhc3MtbWVtYmVycyI6Mn0sImVudiI6e319fQ==)

[Demo reduced to bare minimum showing false positive (demo2)](https://eslint.org/demo/#eyJ0ZXh0IjoiY2xhc3MgRm9vIHtcbiAgZm9vKCkge31cbiAgLyogZm9vICovXG5cbiAgYmFyKCkge31cbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibGluZXMtYmV0d2Vlbi1jbGFzcy1tZW1iZXJzIjoyfSwiZW52Ijp7fX19). This one proves that any block comment between class members break the rule

What's interesting: Moving the `/* eslint-enable class-methods-use-this */` from line 9 to line 10 (so that the padding line is above the `/* eslint-enable */` directive) solves the problem (another demo with no error: [demo1-no-error](https://eslint.org/demo/#eyJ0ZXh0IjoiY2xhc3MgRm9vIHtcbiAgLyogZXNsaW50LWRpc2FibGUgY2xhc3MtbWV0aG9kcy11c2UtdGhpcyAqL1xuICAvKipcbiAgICogQHJldHVybnMge0FycmF5fVxuICAgKi9cbiAgZ2V0Rm9vKCkge1xuICAgIHRocm93IG5ldyBFcnJvcignQ2hpbGQgY2xhc3Mgc2hvdWxkIHJlaW1wbGVtZW50IHRoaXMgbWV0aG9kJyk7XG4gIH1cblxuICAvKiBlc2xpbnQtZW5hYmxlIGNsYXNzLW1ldGhvZHMtdXNlLXRoaXMgKi9cbiAgLyoqXG4gICAqIEByZXR1cm5zIHt1bmRlZmluZWR9XG4gICAqL1xuICBkb1NvbWV0aGluZygpIHt9XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImxpbmVzLWJldHdlZW4tY2xhc3MtbWVtYmVycyI6Mn0sImVudiI6e319fQ==), and [demo2-no-error](https://eslint.org/demo/#eyJ0ZXh0IjoiY2xhc3MgRm9vIHtcbiAgZm9vKCkge31cblxuICAvKiBmb28gKi9cbiAgYmFyKCkge31cbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibGluZXMtYmV0d2Vlbi1jbGFzcy1tZW1iZXJzIjoyfSwiZW52Ijp7fX19))

**What did you expect to happen?**
I think the rule should not report an error on line 14, especially since there's a blank line 10 which logically meets rule requirements.
Placement of `/* eslint-* */` directives should not affect whether the rule reports error or not if there's  one empty line between members.

**What actually happened? Please include the actual, raw output from ESLint.**
Rule reports error in line 14, column 3 (see [demo1](https://eslint.org/demo/#eyJ0ZXh0IjoiY2xhc3MgRm9vIHtcbiAgLyogZXNsaW50LWRpc2FibGUgY2xhc3MtbWV0aG9kcy11c2UtdGhpcyAqL1xuICAvKipcbiAgICogQHJldHVybnMge0FycmF5fVxuICAgKi9cbiAgZ2V0Rm9vKCkge1xuICAgIHRocm93IG5ldyBFcnJvcignQ2hpbGQgY2xhc3Mgc2hvdWxkIHJlaW1wbGVtZW50IHRoaXMgbWV0aG9kJyk7XG4gIH1cbiAgLyogZXNsaW50LWVuYWJsZSBjbGFzcy1tZXRob2RzLXVzZS10aGlzICovXG5cbiAgLyoqXG4gICAqIEByZXR1cm5zIHt1bmRlZmluZWR9XG4gICAqL1xuICBkb1NvbWV0aGluZygpIHt9XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImxpbmVzLWJldHdlZW4tY2xhc3MtbWVtYmVycyI6Mn0sImVudiI6e319fQ==) and [demo2](https://eslint.org/demo/#eyJ0ZXh0IjoiY2xhc3MgRm9vIHtcbiAgZm9vKCkge31cbiAgLyogZm9vICovXG5cbiAgYmFyKCkge31cbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibGluZXMtYmV0d2Vlbi1jbGFzcy1tZW1iZXJzIjoyfSwiZW52Ijp7fX19)).

**Comment by sakabar** [comment]
I'm trying on this!

**Comment by sakabar** [comment]
I made PR. https://github.com/eslint/eslint/pull/9680
Cound you review it?

**Comment by sakabar** [comment]
My mommit message doesn't follow guidelines. I'll fix it.

**Comment by sakabar** [comment]
@Aladdin-ADD Hi. Could you review my PR (https://github.com/eslint/eslint/pull/9680) ? All checks have passed.

**Comment by Aladdin-ADD** [comment]
thanks! I can reproduce it.
